### DISCUSSION TITLE
If a specific folder is selected as the default for uploads, it should also filter Library by that folder by default

### DISCUSSION THREAD

**Discussion by justrealmilk** [directus/directus#9280] [Feature Requests]
![image](https://user-images.githubusercontent.com/156681/139609633-1c8c93d9-f11f-46cf-8591-5f1da1a33835.png)

If a field's default upload folder is set, filter by the same folder when opening the _Choose File from Library_ drawer

Example use case:

![image](https://user-images.githubusercontent.com/156681/139609683-cc36bd26-1200-4d02-8e1f-da0c6643da84.png)

User wishes to upload a set of images as a gallery, but to save space and keep storage clean, they wish to select from those same files as opposed to uploading duplicates. 

Once uploads begin to grow into the thousands, expecting a user to filter to the specific files which they immediately uploaded is frustrating, especially as they aren't consistently displayed as the most recently uploaded images. Load time compounds this frustration.

There does not appear to be a way to pre-define a filter for the files drawer, nor is selecting a folder easy and fast.

**Comment by azrikahar** [comment]
> User wishes to upload a set of images as a gallery, but to save space and keep storage clean, they wish to select from those same files as opposed to uploading duplicates.

There were discussions on dedupe-ing assets based on some form of file hash, which should mitigate this, but this concern is definitely understandable.

> There does not appear to be a way to pre-define a filter for the files drawer

Had a look at the codebase, seems like there are some ground works on this already, just requires some "wiring" of folder ID to predefine the filter. 

I believe the end goal of the filtering is much greater/finer, as per the past discussion you opened: #7239, as well as others like #8998, but I'll try to get a PR for the simplistic version first just to get a rudimentary "filter by specified default folder".

>  nor is selecting a folder easy and fast.

Not that there's any outcome regarding this topic at this point in time, but just to share that there are discussions on adding folder selections in Card layout (which is what the file selection uses) as per #4999.

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
